### PR TITLE
Fix #2033 -- clinical filter returning no variants

### LINKED ISSUE #2033: Check clinical filter performance on clinical-db stage
Clinical filter of release 4.19, 4.20 and current master doesn't work any more. (Well, it returns no variants, but if one clicks on "filter" immediately after, it works).

Thankfully (🙏) everything works on prod (version 4.19)

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #2033.

**How to test**:
1. Go to scout stage and install the current master branch.
1. Open a case and go to variants page
1. Click on the clinical filter button: 

image
1. No variants should be returned
1. Install this branch on scout stage
1. Repeat the search and variants should be shown

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/variants/forms.py [hunk 1]
```diff
@@ -105,14 +105,17 @@ class VariantFiltersForm(FlaskForm):
...
+        "Chromosome", [validators.Optional()], choices=CHROMOSOME_OPTIONS, default=""
+    )
     start = IntegerField("Start position", [validators.Optional()])
     end = IntegerField("End position", [validators.Optional()])
     cytoband_start = NonValidatingSelectField("Cytoband start", choices=[])
     cytoband_end = NonValidatingSelectField("Cytoband end", choices=[])
 
     filter_variants = SubmitField(label="Filter variants")
     export = SubmitField(label="Filter and export")
+    clinical_filter = SubmitField(label="Clinical filter")
```
Comment (dnil) [inline scout/server/blueprints/variants/forms.py:118]: Right, but this isn't really supposed to be instantiated, and Cancer does not have one (yet)?

File: scout/server/blueprints/variants/forms.py [hunk 2]
```diff
@@ -105,14 +105,17 @@ class VariantFiltersForm(FlaskForm):
     load_filter = SubmitField(label="Load filter")
     delete_filter = SubmitField(label="Delete filter")
 
-    chrom = SelectField("Chromosome", [validators.Optional()], choices=CHROMOSOME_OPTIONS)
+    chrom = SelectField(
+        "Chromosome", [validators.Optional()], choices=CHROMOSOME_OPTIONS, default=""
```
Comment (dnil) [inline scout/server/blueprints/variants/forms.py:109]: Was that it?!

Comment (northwestwitch) [reply scout/server/blueprints/variants/forms.py:109]: It's the default option with All chromosomes, look at line 33 :
`CHROMOSOME_OPTIONS = [("", "All")] + [(chrom, chrom) for chrom in CHROMOSOMES]`

File: scout/server/blueprints/variants/forms.py (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/variants/forms.py:118]: This is not really affecting the query I guess, shall I revert this code?

Comment (dnil) [reply scout/server/blueprints/variants/forms.py:118]: I think it would be less confusing, as long as Cancer does not have one,  or what do you think?

Comment (northwestwitch) [reply scout/server/blueprints/variants/forms.py:118]: I can remove it!

File: scout/server/blueprints/variants/forms.py (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/variants/forms.py:109]: > It's the default option with All chromosomes, look at line 33 :
> `CHROMOSOME_OPTIONS = [("", "All")] + [(chrom, chrom) for chrom in CHROMOSOMES]`

Yeah, agree, thats the way it should be - just dont quite see the connection to the clinical filter, which basically has its own set of parameters. But ah - perhaps not for chromosome?

Comment (northwestwitch) [reply scout/server/blueprints/variants/forms.py:109]: Nope, chrom was None

File: scout/server/blueprints/variants/forms.py (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/variants/forms.py]: Done!

File: scout/server/blueprints/variants/forms.py (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/variants/forms.py:109]: One could also set that explicitly in the clinical filter dict. But this deals with multiple potential issues, so lets leave it at that! Well found!

**Review by dnil — APPROVED** [review]
:+1:

**Pull request merged**